vlib/v/parser/tests/map_init.out

```diff
@@ -1,4 +1,4 @@
-vlib/v/parser/tests/map_init.vv:3:19: error: `}` expected, explicit `map` initialization does not support parameters
+vlib/v/parser/tests/map_init.vv:3:19: error: `}` expected; explicit `map` initialization does not support parameters
     1 | fn main() {
     2 |     a := map[string]{}
     3 |     b := map[string]{cap: 10}
```

Comment:
@UweKrueger Does it matter that this is `map[string]{...}` and not `map[string]some_other_type{...}`? Seems like that's the bigger problem in this case tbh

Reply:
Indeed, the parser should not even accept the `a := ...` line. This has nothing to do with my changes, but should be fixed, too. I'll have a look at this...

Reply:
Indeed. Thanks!

Reply:
I've added a check for the value type of a `map` not to be `void` together with some test cases for that.